content/References/dRAIDprimer.md

```diff
@@ -81,48 +90,97 @@ This is essentially a resilver, but data redistributes across the pool to meet t
+dRAID redundancy groups can be much wider than RAIDz vdevs and still have the same level of redundancy. dRAID variables in estimating layouts are:
+
+* Parity (P) in each redundancy group
+* Data disk (D) in each redundancy group
+* Spares (S)
+* Total number of drives, children (C), in the dRAID vdev
+
+When configuring dRAID layouts consider the given number of disks and compare the capacity of those disks laid out in RAIDz and dRAID configurations. The [ZFS Capacity Calculator]({{< relref "/References/ZFSCapacityCalculator.md" >}}) and [ZFS Capacity Graph]({{< relref "/References/ZFSCapacityGraph.md" >}}) allow you to make the comparison.
+
 TrueNAS uses this formula to estimate the total capacity of a dRAID data vdev:
 
 {{< katex >}}
 Capacity = (C - S) \cdot \left(\frac{D}{{D + P}}\right) \cdot DS
 {{< /katex >}}
 
-Where *C* is the children number, *S* is the distributed hot spare number, *D* is the data device number, *P* is the parity level, and *DS* is the minimum disk size common to all disks in the vdev.
+Where:
+* *C* is the children number (total number of drives in the layout).
+* *S* is the distributed hot spare number.
+* *D* is the data device number in the redundancy group.
+* *P* is the parity level in the redundancy group.
+* *DS* is the minimum disk size common to all disks in the vdev.
 
 For example, setting up a single dRAID1 vdev layout with matching 1.82 TiB disks, 8 data devices, 1 distributed hot spare, and 10 children results in an estimated 14.58 TiB (rounded) total capacity:
 
 {{< katex >}}
 (10 - 1) \cdot \left(\frac{8}{{8 + 1}}\right) \cdot 1.82 = 9 \cdot 0.89 \cdot 1.82 = 14.5782
 {{< /katex >}}
 
-This formula doesn't account for additional metadata reservations, so the total estimated capacity is likely to be slightly lower.
+This formula does not account for additional metadata reservations, so the total estimated capacity is likely to be slightly lower.
+
+## dRAID Visualizer
+
+See how dRAID lays out and shuffles data.
+Specify any valid dRAID layout and see it both in a pre-shuffled and a post-shuffled state.
+
+{{< draid >}}
+
+<div style="text-align: right; font-size: smaller; padding-top: 1em;">
+    dRAID Visualizer originally created by <a href="https://jro.io/truenas/draid/">Jason Rose</a>.
+</div>
+
+## Caveats
+
+dRAID layouts require more testing before we make layout recommendations on the number of spares or parity levels.
+When using dRAID, consider these factors:
+
+* dRAID does not support partial-strip writes.
+
+* Small-block writes to dRAID vdev are padded with zeros to span a redundancy group, which creates a lot of wasted space.
+  Include special allocations class vdevs (i.e., L2ARCs, SLOGs, and metadata) in a dRAID layout to handle this type of IO.
+
+* You cannot add virtual spares after creating the vdev because spare disks are distributed into the dRAID vdev.
+
+* dRAID performance matches that of an equivalent RAIDz pool.
```

Comment:
I do not know that we have sufficient empiral data that make any performance claims at this point. dRAID performancce _should_ scale similarly to RAIDz, but until performance can back that up with data I think we should avoid making authoritive statesments related to performance.

Reply:
Modified this statement.